File: app/src/main/java/com/san/moviedbclientmvc/common/di/ActivityCompositionRoot.java
package com.san.moviedbclientmvc.common.di;

import androidx.fragment.app.FragmentActivity;

import com.san.moviedbclientmvc.networking.MoviedbApiService;
import com.san.moviedbclientmvc.common.dialogs.DialogsEventBus;

public class ActivityCompositionRoot {

    private final CompositionRoot mCompositionRoot;
    private final FragmentActivity mActivity;

    public ActivityCompositionRoot(CompositionRoot compositionRoot, FragmentActivity activity) {
        mCompositionRoot = compositionRoot;
        mActivity = activity;
    }

    public FragmentActivity getActivity() {
        return mActivity;
    }

    public MoviedbApiService getMoviedbApiService() {
        return mCompositionRoot.getMovieApiService();
    }

    public DialogsEventBus getDialogsEventBus() {
        return mCompositionRoot.getDialogsEventBus();
    }
}

File: app/src/main/java/com/san/moviedbclientmvc/common/di/ActivityCompositionRoot.kt
package com.san.moviedbclientmvc.common.di

import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentTransaction
import com.san.moviedbclientmvc.common.dialogs.DialogsEventBus

class ActivityCompositionRoot (compositionRoot: CompositionRoot,
                                      activity: FragmentActivity) {

    private val mCompositionRoot: CompositionRoot = compositionRoot
    private val mActivity: FragmentActivity = activity

    fun getActivity(): FragmentActivity = mActivity

    fun getMoviedbApiService() = mCompositionRoot.getMovieApiService()

    fun getDialogsEventBus(): DialogsEventBus? = mCompositionRoot.getDialogsEventBus()
}